public/js/components/modal.jsx

```diff
@@ -10,28 +10,34 @@ module.exports = React.createClass({
   displayName: 'Modal',
 
   propTypes: {
-    children: React.PropTypes.array.required,
-    close: React.PropTypes.func,
-    showModal: React.PropTypes.bool,
+    children: React.PropTypes.object.isRequired,
+    handleClose: React.PropTypes.func,
     title: React.PropTypes.string,
   },
 
-  getDefaultProps: function() {
-    return {
-      showModal: true,
-    };
+  onClose: function(e) {
+    var targetClassName = e.target.getAttribute('class') || '';
+    var classes = targetClassName.split(' ');
+    // Only deal with closing the window if the event
+    // came from the backdrop or the close link.
+    if (classes.length > 0 &&
+        (classes.indexOf('modal') > -1 ||
+         classes.indexOf('close') > -1)) {
+      e.preventDefault();
+      e.stopPropagation();
+      this.props.handleClose();
+    }
```

Comment:
for debugging it might be helpful to log in an else statement that the close handler was not called